src/llm/language_model/legacy/legacy_executor.cpp

```diff
@@ -0,0 +1,87 @@
+//*****************************************************************************
+// Copyright 2025 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "legacy_executor.hpp"
+#include "servable.hpp"
+
+namespace ovms {
+LegacyExecutor::LegacyExecutor(std::shared_ptr<ov::genai::LLMPipeline> pipe) {
+    this->pipe = std::move(pipe);
+}
+
+bool LegacyExecutor::hasRequests() {
+    return (requests.size() > 0);
+}
+
+bool LegacyExecutor::requestsQueueSize() {
+    return requests.size();
+}
+
+void LegacyExecutor::processRequest() {
+    OVMS_PROFILE_FUNCTION();
+    requests.front()->results = pipe->generate(requests.front()->inputIds, requests.front()->apiHandler->createGenerationConfig(), requests.front()->textStreamer);
```

Comment:
is it really good API in terms of of LegacyExecutor class user? It needs to remember, that calling processRequest() when internal state is that requests is empty will cause segfault. It is not even noted anywhere in the code.

At minimum I'd recomment adding comment here and also in the legacy executor header, so we never call this function on empty requests container

Consider changing the API so that processRequest returns at error when internal state is abormal at the moment of calling it?

@mzegla @atobiszei

Reply:
I suppose it's only called inside `if (legacyExecutor->hasRequests())` block, but I guess I won't hurt us if we add such check here and return internal error in case it's called without requests.